### ISSUE TITLE
Kiali config prometheusNamespace istioctl 1.4.0 not working

### ISSUE THREAD

**Issue by alexsotocx** [istio/istio#19064]
**Bug description**

Kiali configuration for prometheus namespace is not working, the configuration is just ignored. 

I'm trying to use an already installed prometheus and I don't have any option apart from the namespace setting to set the url as coded here:

https://github.com/istio/installer/blob/master/istio-telemetry/kiali/templates/configmap.yaml#L41

I haved tried the next things using the next configurations:

```yaml
# values.yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  values:
    prometheus:
      enabled: false
    kiali:
      dashboard:
        grafanaURL: http://monitoring-grafana.monitoring
      enabled: true
      createDemoSecret: true
      prometheusNamespace: monitoring
```

**Running**: `istioctl manifest generate --verbose --log_output_level all:debug -f values.yaml > out.yaml`

**Generates**:

```yaml
# ......
# kiali configmap
apiVersion: v1
kind: ConfigMap
metadata:
  name: kiali
  namespace: istio-system
  labels:
    app: kiali
    release: istio
data:
  config.yaml: |
    istio_component_namespaces:
      grafana: istio-system
      tracing: istio-system
      pilot: istio-system
      prometheus: istio-system
    istio_namespace: istio-system
    deployment:
      accessible_namespaces: ['**']
    server:
      port: 20001
      web_root: /kiali
    external_services:
      istio:
        url_service_version: http://istio-pilot.istio-system:8080/version
      tracing:
        url: 
      grafana:
        url: http://monitoring-grafana.monitoring
      prometheus:
        url: http://prometheus.istio-system:9090 # <=================== See here 
# ......
```

Also tried to set the "global" values, but failed:

```
istioctl manifest generate --verbose --log_output_level all:debug -f values.yaml > out.yaml
input file failed validation with the following errors: unknown field "prometheusNamespace" in v1alpha1.GlobalConfig

Original YAML:
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  values:
    global:
      prometheusNamespace: monitoring
```

Also tried to use the unvalidatedValues, no error but nothing changed:

```
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  unvalidatedValues:
    global:
      prometheusNamespace: monitoring
```

**Expected behavior**

To set the namespace or to provide a way to change the url

**Steps to reproduce the bug**

1. Using istio-1.4.0 try to install with the configuration provided

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```istio git:(MPI-116-prometheus-grafana) ✗ istioctl version --remote
client version: 1.4.0
control plane version: 1.4.0
data plane version: 1.4.0 (7 proxies)
➜  istio git:(MPI-116-prometheus-grafana) ✗ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.7", GitCommit:"8fca2ec50a6133511b771a11559e24191b1aa2b4", GitTreeState:"clean", BuildDate:"2019-09-18T14:47:22Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.7", GitCommit:"8fca2ec50a6133511b771a11559e24191b1aa2b4", GitTreeState:"clean", BuildDate:"2019-09-18T14:39:02Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
➜  istio git:(MPI-116-prometheus-grafana) ✗ helm version
Client: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
➜  istio git:(MPI-116-prometheus-grafana) ✗ ```
```

**How was Istio installed?**

With this guide https://istio.io/docs/setup/getting-started/#download

**Environment where bug was observed (cloud vendor, OS, etc)**

* Mac Os Catalina
* Docker-Kubernetes

**Comment by richardwxn** [comment]
@alexsotocx 
For your first CR yaml, move the prometheusNamespace field to values.global, then run your original istioctl command with --force flag to bypass the validation error
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  values:
    prometheus:
      enabled: false
    global:
      prometheusNamespace: monitoring
    kiali:
      dashboard:
        grafanaURL: http://monitoring-grafana.monitoring
      enabled: true
      createDemoSecret: true
```

And for your second yaml with unvalidated values, that's because you forgot to set values.kiali.enabled to true as above.
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  values:
    kiali:
      enabled: true
  unvalidatedValues:
    global:
      prometheusNamespace: monitoring
```

**Comment by alexsotocx** [comment]
Thanks @richardwxn i Will try in a moment and post my results 👍

**Comment by alexsotocx** [comment]
None one of them works. 
Here the results:
# First
```bash
➜  istio git:(istioctl-config) ✗ cat values.yaml 
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  telemetry:
    enabled: true
  values:
    prometheus:
      enabled: false
    global:
      prometheusNamespace: monitoring
    kiali:
      dashboard:
        grafanaURL: http://monitoring-grafana.monitoring
      enabled: true
      createDemoSecret: true
      prometheusNamespace: monitoring
➜  istio git:(istioctl-config) ✗ istioctl manifest generate --verbose   --force -f values.yaml | grep prometheus: -a1 -b1
Proceeding despite the following validation errors: 
unknown field "prometheusNamespace" in v1alpha1.GlobalConfig
Proceeding despite the following validation errors: 
unknown field "prometheusNamespace" in v1alpha1.GlobalConfig
Proceeding despite the following validation errors: 
unknown field "prometheusNamespace" in v1alpha1.GlobalConfig
270596-      pilot: istio-system
270622:      prometheus: istio-system
270653-    istio_namespace: istio-system
--
--
270946-        url: http://monitoring-grafana.monitoring
270996:      prometheus:
271014-        url: http://prometheus.istio-system:9090
➜  istio git:(istioctl-config) ✗```

**Comment by richardwxn** [comment]
@alexsotocx oh I realized that I tested with latest master not 1.4.0, sry for the confusion. The yamls I provided are correct way to go with, but there was a bug for 1.4.0 so it didn't work for you. The fix was merged in master already, and should be available for 1.4.1.

**Comment by alexsotocx** [comment]
Ohh Thanks  :) I will wait so
